src/app/auth.guard.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@angular/core';
+import {CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot, UrlTree, Router} from '@angular/router';
+import { Observable } from 'rxjs';
+import {AuthService} from './auth.service';
+import {UserService} from './user.service';
+import {map} from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuard implements CanActivate {
+
+  constructor(private auth: AuthService,
+              private router: Router,
+              private user: UserService) {  }
+
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    if (this.auth.isLoggedIn) {
+      return true;
+    }
+    return this.user.isLoggedIn().pipe(map(res => {
+      if (res.status) {
```

Comment:
Either naming is wrong, or implementation is bad. What does res.status return? User object? Or HTTP status?